delivery_correos_express/models/delivery_carrier.py

```diff
@@ -0,0 +1,260 @@
+                "ref": picking.name,
+                "refCliente": picking.name,
+                "fecha": fields.Datetime.now().strftime("%d%m%Y"),  # mandatory
+                "contacOtrs": "",
+                "telefOtrs": "",
+                "emailOtrs": "",
+                "observac": "",
+                "numBultos": number_of_packages or 1,  # mandatory
+                "kilos": "%.3f" % (picking.shipping_weight or 1),  # mandatory
+                "volumen": "",
+                "alto": "",
+                "largo": "",
+                "ancho": "",
+                "producto": "93",  # ePaq24  # mandatory
+                "portes": "P",  # mandatory
+                "reembolso": "",
+                "entrSabado": "",
+                "seguro": "",
+                "numEnvioVuelta": "",
+                "listaBultos": package_list,
+                "codDirecDestino": "",
+                "password": "",
+                "listaInformacionAdicional": [
+                    {"tipoEtiqueta": self.correos_express_label_type, "etiquetaPDF": ""}
+                ],
+            },
+            **self._get_correos_express_sender_info(picking),
+            **self._get_correos_express_receiver_info(picking),
+        )
+
+    def correos_express_send_shipping(self, pickings):
+        correos_express_request = CorreosExpressRequest(self)
+        result = []
+        is_pdf = self.correos_express_label_type != "2"
+        for picking in pickings:
+            vals = self._prepare_correos_express_shipping(picking)
+            response = correos_express_request.create_shipment(vals)
+            if not response:
+                result.append(vals)
+                continue
+            vals.update(
+                {
+                    "tracking_number": response.get("datosResultado", ""),
+                    "exact_price": 0,
+                }
+            )
+            attachments = []
+            if response.get("etiqueta"):
+                # To decode the label with Base64 we need to decode it first
+                # to binary and afterwards decode again to transform it
+                # into a PDF or text document
+                attachments = [
+                    (
+                        "correos_express_{}_{}.{}".format(
+                            response.get("datosResultado", ""),
+                            index + 1,
+                            "pdf" if is_pdf else "txt",
+                        ),
+                        (
+                            base64.b64decode(
+                                base64.b64decode(label.get("etiqueta1", ""))
+                            )
+                            if is_pdf
+                            else label.get("etiqueta2", "")
+                        ),
+                    )
+                    for index, label in enumerate(response.get("etiqueta"))
+                ]
+            picking.message_post(body=_(""), attachments=attachments)
```

Comment:
```suggestion
                attachments = [
                    (
                        "correos_express_{}_{}.{}".format(
                            response.get("datosResultado", ""),
                            index + 1,
                            "pdf" if is_pdf else "txt",
                        ),
                        (
                            base64.b64decode(
                                base64.b64decode(label.get("etiqueta1", ""))
                            )
                            if is_pdf
                            else label.get("etiqueta2", "")
                        ),
                    )
                    for index, label in enumerate(response.get("etiqueta"))
                ]
            picking.message_post(body=_(""), attachments=attachments)
```
Another option would be to create the attachment directly (example: https://github.com/OCA/delivery-carrier/blob/13.0/delivery_ups_oca/models/delivery_carrier.py#L124-L132)

Reply:
Does this post it in the chatter at the side? I prefer having it show up there

Reply:
This comment is only a suggestion although in both cases the attachment will be created and will remain linked to the record (the position of the chatter depends on the user's configuration).

Reply:
I think I prefer letting Odoo handle the attachment creation with the `message_post`